dre-service/src/main/java/io/cdap/re/RulesDB.java

```diff
@@ -152,7 +164,9 @@ void createRule(RuleRequest rule) throws RuleAlreadyExistsException, IOException
     long currentTime = getCurrentTime();
 
     Collection<Field<?>> ruleFields = new ArrayList<>();
-    ruleFields.add(Fields.stringField(ID_COL, rule.getId()));
+    ruleFields.add(Fields.stringField(NAMESPACE_COL, namespaceName));
```

Comment:
this is done in several places, it's worth extracting it out into a `createKeyFields(NamespacedId id)` method

Reply:
I added this method in the latest commit